.pre-commit-config.yaml

```diff
@@ -0,0 +1,92 @@
+repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.0.1
```

Comment:
I would suggest updating the "rev" tags to the latest releases on their Github pages, unless there's a reason to hold the versions back.

Reply:
Good point. I did update most versions to latest. With two exceptions:
* I only updated flake8 to 5.0.4 since 6.0 started complaining about missing imports when the imports are only used in variable type comments. Once we are finally past python 3.5 we can make that update.
* Only updated black to 22.12.0 to avoid some changes 23.1 would be making to keep the codebase in sync with the Orca Collective fork until all the back-porting is done.